src/lib/serviceRoutinesV2/postBatchQuery.cpp

```diff
@@ -69,6 +69,16 @@ std::string postBatchQuery
   qcrP->fill(bqP);
   bqP->release();  // qcrP just 'took over' the data from bqP, bqP no longer needed
 
+  //
+  // This request does not support (it ignores) the URI parameter 'metadata'.
+  // Instead the metadata filter comes inside the payload.
+  // To pass the metadata filter to the rendering function, the same mechanism as for URI param is used here.
+  // This is a little bit ugly as we 'fool' the render method to believe that the filter came via URI param.
+  // However, the implementation is cleaner and no more code is needed for this, just to set the
+  // URI param with the value of the metadata filter from the payload.
+  //
+  ciP->uriParam[URI_PARAM_METADATA] = bqP->metadataV.toString();
```

Comment:
Yeah, what is done here is a little ugly.
The URI param (if present) is overwritten with a stringifiction of the metadata vector from the payload and sent to the next function (the one you indicate), where the stringification is parsed into a vector again ... Ugly, however even more ugly may be to add a parameter to accept either a string-vector or a string ...   This is more or less explained in the comment right above the line in the source code